### PR TITLE
HPCC-17005 Remove dafilesrv execute functionality.

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

## Testing:

Tested with unit tests, that were added ahead of this issue, in part to ensure that there were no side-effects elsewhere in the remote/dafilesrv code.

### CODE REVIEW

**Comment by jakesmith** [comment]
@mckellyln - please review

**Comment by mckellyln** [comment]
@jakesmith what is supposed to happen if an older client tries to ask this newer dafilesrv to cmdExec a script ?
I think it would be handled as a cmdUnknown now ?  Which means the client would get an "Invalid cmd" msg back ?

**Comment by richardkchapman** [comment]
@jakesmith typo in commit message

**Comment by jakesmith** [comment]
@mckellyln - please review added commit.

File: common/remote/sockfile.cpp [hunk 1]
```diff
@@ -4752,44 +4718,14 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
...
-        if (outsz)
-            reply.append(outbuf);
+        StringAttr cmdLine;
+        msg.read(cmdLine);
+        // NB: legacy remoteExec used to simply pass error code and buffer back to caller.
+        VStringBuffer errMsg("Remote command execution no longer supported. Trying to execute cmdline=%s", cmdLine.get());
+        WARNLOG("%s", errMsg.str());
+        size32_t outSz = errMsg.length()+1; // reply with null terminated string
+        // reply with error code -1
+        reply.append((unsigned)-1).append((unsigned)0).append(outSz).append(outSz, errMsg.str());
```
Comment (mckellyln) [inline common/remote/sockfile.cpp]: Why outSz twice ?

**Review by mckellyln — APPROVED** [review]
Approved.

**Comment by richardkchapman** [comment]
@jakesmith Can you squash before I merge

**Comment by jakesmith** [comment]
@richardkchapman - done.

**Pull request merged**